### ISSUE TITLE
about clipping_and_adding_noise

### ISSUE THREAD

**Issue by WJYuuuu** [JeffffffFu/Awesome-Differential-Privacy-and-Meachine-Learning#11]
在文件 optimizer 中的clipping_and_adding_noise.py文件中，那个裁剪好像是对模型参数进行裁剪，我没有太看明白是什么裁剪技巧。
`            per_data_parameters_grad_dict[key] /= max(1,model_parameter_grad_norm/max_norm)
`
这里面的per_data_parameters_grad_dict是从模型参数里得到的`        
for key,var in params.items():   # params.items()
            model_parameter_grad_norm+=var.data.norm(2)**2  
            per_data_parameters_grad_dict[key]=var.clone().detach()
`
